ext/http/lib.rs

```diff
@@ -244,13 +245,17 @@ fn prepare_next_request(
   let url = req_url(&req, scheme, addr)?;
 
   let is_websocket = is_websocket_request(&req);
-  let has_body = if let Some(exact_size) = req.size_hint().exact() {
+  let maybe_has_body = if let Some(exact_size) = req.size_hint().exact() {
```

Comment:
Maybe `can_have_body`? The `maybe_foo` idiom is normally used to indicate it's an Option. Using it for a boolean is kind of confusing.

I'd perhaps flip the logic a little so size_hint() is only called when it's needed. E.g.:
```rust
let can_have_body = !matches!(*req.method(), Method::GET | Method::HEAD);
let has_body = is_websocket || can_have_body && Some(0) != req.size_hint().exact();
```
Not that the current code is wrong or inefficient but you have to pay close attention to the logic to figure out when its result is actually used.

Reply:
Yeah, that makes sense. Updated to your logic.